driver/src/driver/scheduler.rs

```diff
@@ -0,0 +1,274 @@
+use super::stablex_driver::StableXDriver;
+use anyhow::{anyhow, Context, Result};
+use log::error;
+use log::info;
+use std::thread;
+use std::time::{Duration, SystemTime, SystemTimeError};
+
+const BATCH_DURATION: Duration = Duration::from_secs(300);
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+struct BatchId(u64);
+
+impl BatchId {
+    pub fn current(now: SystemTime) -> std::result::Result<Self, SystemTimeError> {
```

Comment:
None of these methods need to be `pub` as `BatchId` is only used in this module right?